### ISSUE TITLE
`warnings.UserWarning` not suppressed when importing modules to parse their syntax trees.

### ISSUE THREAD

**Issue by emcd** [pylint-dev/pylint#7906]
### Bug description

The following code:
```python
''' Minimal reproducer for setuptools warning. '''
import setuptools
import pip
```
produces the following warning when scanned with `pylint`:
```
.../site-packages/_distutils_hack/__init__.py:33: UserWarning: Setuptools is replacing distutils.
  warnings.warn("Setuptools is replacing distutils.")
```
This output is confusing as it is triggered by the code being scanned and not by anything that Pylint uses internally.

### Command used

```shell
pylint --report=no --score=no --disable=unused-import bugs/pylint/distutils_warning.py
```

### Pylint output

```shell
.../site-packages/_distutils_hack/__init__.py:33: UserWarning: Setuptools is replacing distutils.
  warnings.warn("Setuptools is replacing distutils.")
```

### Expected behavior

I would expect no `UserWarning` for something related to Setuptools while scanning code with Pylint.

While I think that this is mostly a toxic interaction between Setuptools and Pip imports, Pylint (or Astroid) could suppress these warnings by wrapping the imports needed for AST parsing with the `warnings.catch_warnings` context manager or something along those lines.

### Pylint version

```shell
pylint --version
pylint 2.15.7
astroid 2.12.13
Python 3.7.15 (default, Oct 15 2022, 11:29:20)
[GCC 9.4.0]
```
